lang/java/reef-runtime-local/src/main/java/org/apache/reef/runtime/local/driver/ResourceManager.java

```diff
@@ -199,41 +203,87 @@ void onResourceLaunchRequest(
   }
 
   /**
+   * Check if the racks used in the request are the ones available in the local runtime, otherwise throw an exception
+   * @param rackNames the "user defined" rack names to simulate rack awareness in the local runtime
+   */
+  private void validateRackNames(final List<String> rackNames) {
+    for (final String rackName : rackNames) {
+      if (!availableRacks.contains(rackName)) {
+        throw new IllegalArgumentException("Rack requested for Evaluators does not exist in the local runtime: "
+            + rackName + ", available racks are: " + availableRacks.toString());
+      }
+    }
+  }
+
+  private List<String> getRackNamesOrDefault(final List<String> rackNames) {
+    return CollectionUtils.isNotEmpty(rackNames) ? rackNames : Arrays.asList(RackNames.DEFAULT_RACK_NAME);
```

Comment:
I agree that is expensive, but is the local runtime, it's just for debugging purposes, we won't request evaluators for 100 racks, neither configure 100 available racks...
Is a validation to check if the rack in the evaluator request is available in the local runtime. If not, then there was a misconfiguration from the developer, we trigger that exception...